### PR TITLE
Add logit to Register web_App

### PR DESCRIPTION
**Pull request by felixtheflex**
### Context

Add logging capabilities using Logit

### Changes proposed in this pull request

The proposal is not to change anything; but to add new functionality and capabilities

### CODE REVIEW

File: terraform/modules/paas/main.tf [hunk 1]
```diff
@@ -59,3 +59,11 @@ resource cloudfoundry_route web_app_route {
   space    = data.cloudfoundry_space.space.id
   hostname = local.web_app_name
 }
+
+resource "cloudfoundry_user_provided_service" "logging" {
+  count            = var.logging
```
Comment (saliceti) [inline terraform/modules/paas/main.tf:64]: Do we need a `count` here? I believe there will be only one service

File: terraform/modules/paas/main.tf [hunk 2]
```diff
@@ -59,3 +59,11 @@ resource cloudfoundry_route web_app_route {
...
+  name             = var.logging_name
```
Comment (saliceti) [inline terraform/modules/paas/main.tf:65]: `logging` is not very explicit. Could we replace this with something like `logging_service_name`?

File: terraform/modules/paas/variables.tf [hunk 3]
```diff
@@ -19,6 +19,10 @@ variable web_app_memory { default = 512 }
 variable worker_app_instances { default = 1 }
 variable worker_app_memory { default = 512 }
 
+variable logging {}
```
Comment (saliceti) [inline terraform/modules/paas/variables.tf:22]: This value can be calculated in the module, there is no need to pass it as an input, so we can simplify the interface.
Can we add a local variable `logging_service_name` instead?

File: terraform/modules/paas/main.tf (hunk 2 above)
Comment (felixtheflex) [reply terraform/modules/paas/main.tf]: done...

File: terraform/terraform.tf [hunk 4]
```diff
@@ -31,4 +31,8 @@ module paas {
   web_app_memory              = var.paas_web_app_memory
   worker_app_instances        = var.paas_worker_app_instances
   worker_app_memory           = var.paas_worker_app_memory
+
+  logging               = var.pass_logging
+  logging_endpoint_port = var.pass_logging_endpoint_port
+  logging_service_name          = var.pass_logging_service_name
```
Comment (defong) [inline terraform/terraform.tf:37]: formatting

File: terraform/workspace-variables/qa.tfvars [hunk 5]
```diff
@@ -10,3 +10,6 @@ paas_web_app_instances           = 1
 paas_web_app_memory              = 512
 paas_worker_app_instances        = 1
 paas_worker_app_memory           = 512
+
```
Comment (defong) [inline terraform/workspace-variables/qa.tfvars:13]: formating

File: terraform/terraform.tf (hunk 4 above)
Comment (defong) [reply terraform/terraform.tf:37]: also what is pass?

File: terraform/workspace-variables/qa.tfvars (hunk 5 above)
Comment (defong) [reply terraform/workspace-variables/qa.tfvars:13]: also what is pass?

File: terraform/modules/paas/main.tf [hunk 6]
```diff
@@ -59,3 +59,11 @@ resource cloudfoundry_route web_app_route {
   space    = data.cloudfoundry_space.space.id
   hostname = local.web_app_name
 }
+
+resource "cloudfoundry_user_provided_service" "logging" {
```
Comment (saliceti) [inline terraform/modules/paas/main.tf:63]: Please remove the double quotes

**Comment by defong** [comment]
[[M] Send logs to logit](https://trello.com/c/0UirUx1x/35-m-send-logs-to-logit)

**Review by saliceti — APPROVED** [review]
OK

File: terraform/modules/paas/variables.tf [hunk 7]
```diff
@@ -19,6 +19,9 @@ variable web_app_memory { default = 512 }
 variable worker_app_instances { default = 1 }
 variable worker_app_memory { default = 512 }
 
+variable logging_endpoint_port {}
```
Comment (vigneshmsft) [inline terraform/modules/paas/variables.tf:22]: can we call it `logit_url` or just `logging_url`?
also can this be moved to `locals` as we are going to just have one logit stack for all envs?

Comment (saliceti) [reply terraform/modules/paas/variables.tf:22]: `log_url` looks fine
We need to keep it as an input variable since this should be considered a secret

Comment (felixtheflex) [reply terraform/modules/paas/variables.tf]: done...

**Review by vigneshmsft — APPROVED** [review]

**Pull request merged**